### PR TITLE
add wire panel and functions to apc

### PR DESCRIPTION
**Pull request by VerinSenpai**
## About the PR
Adds wire panel functionality to apcs to A:) make it a touch more difficult to dismantle them and B:) allow for disabing AI access and disabling access restrictions to APCs, similarly to how air alarms work. My issue right now is that in order to implement the power wire properly, its looking like I may have to make changes to ApcComponent.cs, PowerWireAction.cs, and ApcPowerReceiverComponent.cs. The purpose of this PR is to get feedback on if this is worth pursuing and ideas on ways I can implement the power wire functionality without making large overarching changes or more realistically without me making a mess out of other peoples code.

## Why / Balance
Currently, its a touch too easy in my opinion to dismantle apcs and substations. It makes no sense how you are able to pry electrified components out without insuls. This also parities air alarms now in that you can disable an AI's access to them and allow for anyone to control them by snipping the access and ai access wires.

## Technical details
As of current function for the power wire isn't implemented. Everything else though is implemented through yml changes.

## Media

https://github.com/user-attachments/assets/e775f1da-f0df-4e92-9982-c7773d62da7e

## Requirements

- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes
No breaking changes made... yet?

**Changelog**
:cl:
- add: APC maintenance panels are functional now.

### CODE REVIEW

**Comment by VerinSenpai** [comment]
wrote some code that I'm very displeased with. PowerWireAction being requiring a powered device be apc supplied hurts my soul. I'm lacking a bit in the c# department so any feedback or ideas for a better way to do this would be greatly appreciated.

**Comment by Alpaccalypse** [comment]
I'm in the same boat, so can't help with the C#, unfortunately, but to address your PR description, I do think this is well worth persuing. All your arguments for this ring true and make sense to me.

**Comment by Alpaccalypse** [comment]
I don't know if there is a good reason for this, but if not, it seems related to this draft:

Borgs cannot toggle APCs, which seems like an oversight, as they can interact with almost any other electrical device with a UI. I feel that, at least, engineering borgs should be able to do this. Apologies if this is more appropriate to a separate PR.

**Comment by VerinSenpai** [comment]
> I don't know if there is a good reason for this, but if not, it seems related to this draft:
> 
> Borgs cannot toggle APCs, which seems like an oversight, as they can interact with almost any other electrical device with a UI. I feel that, at least, engineering borgs should be able to do this. Apologies if this is more appropriate to a separate PR.

I took a looksee for this but I'm unable to reproduce it. the only borgs that can't toggle apcs are syndicate ones as far as I can tell. Is it possible this is already fixed and pending an update?

**Comment by VerinSenpai** [comment]
could someone remove the no c# tag for me? I don't know if thats affecting tests or not but most of em are skipped.

**Comment by Alpaccalypse** [comment]
> I took a looksee for this but I'm unable to reproduce it. the only borgs that can't toggle apcs are syndicate ones as far as I can tell. Is it possible this is already fixed and pending an update?

Thanks for looking. I'll let you know if I come across this again, if it still seems relevant, otherwise I'll leave you to focus on what you already have going.

**Comment by VerinSenpai** [comment]
@ArtisticRoomba I know this is a bit out of the way but you wouldn't happen to know if power or wires are being refactored currently would you? asking because I don't like how I'm currently going about disabling the apc when wires are cut. Course I ain't looked at this in a month.

**Comment by ArtisticRoomba** [comment]
> @ArtisticRoomba I know this is a bit out of the way but you wouldn't happen to know if power or wires are being refactored currently would you? asking because I don't like how I'm currently going about disabling the apc when wires are cut. Course I ain't looked at this in a month.

As far as my knowledge goes there's some planned expansion on how machines interact with Pow3r.

Pow3r describes how the station behaves under deficits caused by grid fluctuations, which are generally caused by mismatches between supply and demand. An analog device like a pump, motor, battery charger, would see reduced work or output. Digital devices like most advanced machines, sensors, etc. on station would enter a bootloop until they have enough power to stay on.

But those are general interactions with the powernet. If you're talking about the core tech behind Pow3r I think all of that is fine as we're chosen the [BatteryRampPegSolver](https://docs.spacestation14.com/en/space-station-14/departments/engineering/pow3r.html#batteryramppegsolver) system and all of it is implemented.

TLDR: Core tech of Pow3r isn't seeing a refactor (as far as I'm aware of!!!!!!). However you might want to design your wirepanel interactions with not-shitcode in mind when someone comes in and works on APC equipment networks.

**Comment by VerinSenpai** [comment]
Closing this to get it off the backlog. I'll reapproach it when my understanding of the codebase has improved or if someone else would like to take the mantle >.>
